lib/iris/tests/test_image_json.py

```diff
@@ -0,0 +1,92 @@
+# (C) British Crown Copyright 2016, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+import six
+
+# import iris tests first so that some things can be initialised before
+# importing anything else
+import iris.tests as tests
+
+import codecs
+import json
+import logging
+import os
+import requests
+from collections import deque
+from itertools import chain
+from six.moves.queue import Queue
+from threading import Thread
+
+# maximum number of threads for multi-threading code
+MAXTHREADS = 128
```

Comment:
Do you really need to run 128 parallel HEAD requests? Browsers limit themselves to something like 2-3 connections per host; this seems like overkill.

When I run it with 128, I get a bunch of `ConnectionError(MaxRetryError...` in the exception deque and it takes 22.277s. If I cut threads down to _2_, there are no errors and it runs in 16.276s. Running with 1 thread takes 33.248s, 4 threads is 8.292s and 8 threads is 5.938s. The delta is already diminishing there, so I doubt there is any benefit to bumping it all the way to 128.